pandasql/io.py

```diff
@@ -1,10 +1,33 @@
+import os
+import subprocess
+
 import pandas as pd
 
-from pandasql.core import DataFrame
+from pandasql.core import DB_FILE, SQL_CON, DataFrame, _new_name
+
+SAMPLE_LINES = 1000
+MEMORY_THRESHOLD = 4_000_000_000    # 4 GB
+CHUNKSIZE = 10_000
 
 
 def read_csv(*args, name=None, **kwargs):
-    return DataFrame(pd.read_csv(*args, **kwargs), name=name)
+
+    if 'nrows' in kwargs:
+        raise ValueError('nrows is not supported')
+
+    name = name or _new_name()
+    sql_load = kwargs.pop('sql_load') if 'sql_load' in kwargs else False
+
+    if sql_load:    # read through sqlite
+        return _csv_to_sql(*args, name=name, **kwargs)
+
+    estimated_size = _estimate_pd_memory(args[0])
```

Comment:
This will break if the user specifies the file name as a kwarg. I suggest adding the first argument to `read_csv` explicitly, instead of capturing via `*args` or `**kwargs`.